core/src/test/scala/org/apache/spark/shuffle/sort/io/LocalDiskShuffleMapOutputWriterSuite.scala

```diff
@@ -0,0 +1,223 @@
+ * limitations under the License.
+ */
+
+package org.apache.spark.shuffle.sort.io
+
+import java.io.{ByteArrayInputStream, File, FileInputStream, FileOutputStream}
+import java.math.BigInteger
+import java.nio.ByteBuffer
+
+import org.mockito.Answers.RETURNS_SMART_NULLS
+import org.mockito.ArgumentMatchers.{any, anyInt, anyLong}
+import org.mockito.Mock
+import org.mockito.Mockito.{doAnswer, doNothing, when}
+import org.mockito.MockitoAnnotations
+import org.scalatest.BeforeAndAfterEach
+
+import org.apache.spark.{SparkConf, SparkFunSuite}
+import org.apache.spark.executor.ShuffleWriteMetrics
+import org.apache.spark.network.util.LimitedInputStream
+import org.apache.spark.shuffle.IndexShuffleBlockResolver
+import org.apache.spark.util.ByteBufferInputStream
+import org.apache.spark.util.Utils
+
+class LocalDiskShuffleMapOutputWriterSuite extends SparkFunSuite with BeforeAndAfterEach {
+
+  @Mock(answer = RETURNS_SMART_NULLS) private var blockResolver: IndexShuffleBlockResolver = _
+  @Mock(answer = RETURNS_SMART_NULLS) private var shuffleWriteMetrics: ShuffleWriteMetrics = _
+
+  private val NUM_PARTITIONS = 4
+  private val D_LEN = 10
+  private val data: Array[Array[Int]] = (0 until NUM_PARTITIONS).map {
+    p => (1 to D_LEN).map(_ + p).toArray }.toArray
+
+  private var tempFile: File = _
+  private var mergedOutputFile: File = _
+  private var tempDir: File = _
+  private var partitionSizesInMergedFile: Array[Long] = _
+  private var conf: SparkConf = _
+  private var mapOutputWriter: LocalDiskShuffleMapOutputWriter = _
+
+  override def afterEach(): Unit = {
+    try {
+      Utils.deleteRecursively(tempDir)
+    } finally {
+      super.afterEach()
+    }
+  }
+
+  override def beforeEach(): Unit = {
+    MockitoAnnotations.initMocks(this)
+    tempDir = Utils.createTempDir(null, "test")
+    mergedOutputFile = File.createTempFile("mergedoutput", "", tempDir)
+    tempFile = File.createTempFile("tempfile", "", tempDir)
+    partitionSizesInMergedFile = null
+    conf = new SparkConf()
+      .set("spark.app.id", "example.spark.app")
+      .set("spark.shuffle.unsafe.file.output.buffer", "16k")
+    when(blockResolver.getDataFile(anyInt, anyInt)).thenReturn(mergedOutputFile)
+
+    doNothing().when(shuffleWriteMetrics).incWriteTime(anyLong)
+
+    doAnswer { invocationOnMock =>
+      partitionSizesInMergedFile = invocationOnMock.getArguments()(2).asInstanceOf[Array[Long]]
+      val tmp: File = invocationOnMock.getArguments()(3).asInstanceOf[File]
+      if (tmp != null) {
+        mergedOutputFile.delete
+        tmp.renameTo(mergedOutputFile)
+      }
+      null
+    }.when(blockResolver)
+      .writeIndexFileAndCommit(anyInt, anyInt, any(classOf[Array[Long]]), any(classOf[File]))
+    mapOutputWriter = new LocalDiskShuffleMapOutputWriter(
+      0,
+      0,
+      NUM_PARTITIONS,
+      shuffleWriteMetrics,
+      blockResolver,
+      conf)
+  }
+
+  private def readRecordsFromFile(fromByte: Boolean): Array[Array[Int]] = {
```

Comment:
this method is super confusing to me.  I don't understand the point of `fromByte` -- seems like it would be a lot easier if you just consistently wrote out Ints to the file, and read them back.  Also I think everything becomes a lot simpler if you use DataInputStream